grails-app/controllers/org/pih/warehouse/order/OrderController.groovy

```diff
@@ -623,6 +626,10 @@ class OrderController {
         Order order = Order.get(params.order.id)
         OrderItem orderItem = OrderItem.get(params.orderItem.id)
         ProductSupplier productSupplier = null
+        if (params.validationCode == 'BLOCK') {
```

Comment:
Let's use the enum ValidationCode.BLOCK.

Reply:
I'ts not working with ValidationCode.BLOCK

Reply:
Sorry you'd need to convert params.validationCode to ValidationCode enum. Something like this should work but you'd need to do some error handling for when params.validationCode is null.
```
ValidationCode validationCode = params.validationCode ? params.validationCode as ValidationCode : null
if (validationCode == ValidationCode.BLOCK) { 
```

Other options that might also work ... but I prefer the above approach
```
ValidationCode validationCode = params.validationCode as ValidationCode
if (params.validationCode == ValidationCode.BLOCK.toString()) { 
```
or 
```
ValidationCode validationCode = params.validationCode as ValidationCode
if (params.validationCode == ValidationCode.BLOCK.name()) { 
```